docs/proposals/bearer-token.md

```diff
@@ -0,0 +1,153 @@
+// FileTokenService - reads a token from a static file
+type FileTokenService struct {
+   tokenFile string
+}
+
+func (f FileTokenService) Validate(token string) bool {
+    // read the file from tokenFile
+    // compare the token from the file to the one given
+    savedToken, err := ioutil.ReadFile(tokenFile)
+    if err != nil {
+        return false
+    }
+
+    return token == savedToken
+}
+```
+
+If we have an external service to handle validation something like this would be
+used:
+
+```golang
+type SomeExternalService struct {
+    service SomeAuthService
+}
+
+func NewSomeExternalService(service SomeAuthService) SomeExternalService {
+    return SomeExternalService{service: service}
+}
+
+func (s SomeExternalService) Validate(token string) bool {
+    return s.service.Validate(token)
+
+    // this could also be a simple get the value from the
+    // service, then compare it. Or much more complex if need be.
+    // Basically the idea is the AuthProviders calling Validate
+    // don't really care.
+}
+```
+
+
+Basic questions about the impact to Broker and APBs:
+
+ - How will the broker's behavior change?
+
+With the addition of `BearerAuth` we have expanded the broker's authentication
+mechanisms to 2 ways for authentication. These mechanisms are NOT mutually
+exclusive. The broker supports running all of the authentication mechanisms at
+the same time.
+
+Basically, the middleware handler will pass the request to each of the configured
+`AuthProviders`. The first `AuthProvider` that knows how to handle the auth
+wins.
+
+Assume both basic auth and bearer token are enabled. If a Bearer token header is
+submitted, the basic auth provider will ignore it. The bearer token provider
+will pick it up. All other configured auth providers would be skipped since the
+auth has occurred.
+
+For example, if we had an SSL auth provider, it would have gotten skipped because
+the bearer token provider already processed the authentication.
+
+ - Will this change APBs?
+
+The bearer token will have no affect on the APBs.
+
+ - Will there be any developer impact?
+
+By default auth is disabled when running the broker locally, the broker developer
+will only be impacted if they need to enable authentication while running
+locally. Then it will require modifying the ConfigMap in the
+`deploy-local-dev-changes.yaml` template to enable auth of choice.
+
+They will also need to create directories required by the auth system to
+simulate auth secrets.
+
+The best way to test broker with authentication is to build a new image, `make
+build-image ORG=YOURORG TAG=YOURTAG` and run `make deploy`.
+
+
+
+#### Issues
+
+ - What's the best way to configure different auths?
+
+For example, I want to have a `BasicAuth` that uses the `FileUserService` and a
+`BasicAuth` that uses `DBUserService`, a fictitious service that loads users from a
+database. Today's configuration does not support specifying a service backend
+to a particular `AuthProvider`. Thoughts?
+
```

Comment:
@rthallisey not yet, right now basicauth is tied to FileUserService and bearer auth would be tied to FileTokenService. Just need to decide how we want this in the config file.